main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    result = []
+    my_each do |i|
+      result.push(i) if yield i
+    end
+    result
+  end
+
+  def my_all?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_any?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_none?(*val)
+    result = false
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false if yield(i) }
+    end
+    result
+  end
+
+  def my_count(*)
+    return length unless block_given?
+
+    count = 0
+    my_each do |i|
+      count += 1 if yield i
+    end
+    count
+  end
```

Comment:
# my_count
- [ ] `my_count` returns the number of items in enum through enumeration::range

      Failure/Error: expect(range.my_count).to eq range.count

      NameError:
        undefined local variable or method `length' for 5..50:Range
##
- [ ] `my_count` counts the number of items in enum that are equal to an item if an argument is given

      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)

        expected: 12
             got: 100